packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/ReactViewGroupMapper.kt

```diff
@@ -0,0 +1,113 @@
+import com.facebook.react.uimanager.Spacing
+import com.facebook.react.views.view.ReactViewBackgroundDrawable
+import com.facebook.react.views.view.ReactViewGroup
+
+internal class ReactViewGroupMapper :
+    BaseWireframeMapper<ReactViewGroup, MobileSegment.Wireframe>(),
+    TraverseAllChildrenMapper<ReactViewGroup, MobileSegment.Wireframe> {
+
+    @SuppressLint("VisibleForTests")
+    override fun map(
+        view: ReactViewGroup,
+        mappingContext: MappingContext,
+        asyncJobStatusCallback: AsyncJobStatusCallback
+    ): List<MobileSegment.Wireframe> {
+        val viewGlobalBounds = resolveViewGlobalBounds(
+            view,
+            mappingContext.systemInformation.screenDensity
+        )
+
+        val pixelDensity = mappingContext.systemInformation.screenDensity
+        val backgroundDrawable = view.background as? ReactViewBackgroundDrawable ?: return emptyList()
+
+        val backgroundColor = view.backgroundColor
+        if (backgroundColor == Color.TRANSPARENT) {
+            return emptyList()
+        }
+        val opacity = resolveOpacity(view, view.alpha)
+
+        // view.alpha is the value of the opacity prop on the js side
+        // this should override the alpha value of the background color if it is explicitly 0
+        if (opacity == 0f) {
+            return emptyList()
+        }
+
+        val viewAlpha = resolveAlpha(opacity, backgroundColor)
+
+        val backgroundColorAndAlpha = colorAndAlphaAsStringHexa(
+            color = backgroundColor,
+            alphaAsHexa = viewAlpha
+        )
+
+        val (shapeStyle, border) = resolveRNShapeStyleAndBorder(
+            backgroundDrawable = backgroundDrawable,
+            viewAlpha = viewAlpha,
+            viewColor = backgroundColorAndAlpha,
+            opacity = opacity,
+            pixelDensity = pixelDensity
+        )
+
+        return listOf(
+            MobileSegment.Wireframe.ShapeWireframe(
+                resolveViewId(view),
+                viewGlobalBounds.x,
+                viewGlobalBounds.y,
+                viewGlobalBounds.width,
+                viewGlobalBounds.height,
+                shapeStyle = shapeStyle,
+                border = border
+            )
+        )
+    }
+
+    private fun resolveRNShapeStyleAndBorder(
+        backgroundDrawable: ReactViewBackgroundDrawable?,
+        viewAlpha: Int,
+        viewColor: String,
+        opacity: Float,
+        pixelDensity: Float
+    ):
+            Pair<MobileSegment.ShapeStyle?, MobileSegment.ShapeBorder?> {
+        if (backgroundDrawable == null) {
+            return MobileSegment.ShapeStyle(
+                backgroundColor = viewColor,
+                opacity = viewAlpha,
+            ) to null
+        } else {
+            val cornerRadius = backgroundDrawable.fullBorderRadius
+            val borderColor = resolveBorderColor(backgroundDrawable, opacity)
+            val borderWidth = backgroundDrawable.fullBorderWidth
```

Comment:
On the screenshots you shared I'm under the impression that the border appears bigger in the replay than on emulators. 
Could you check that this works properly?

Reply:
yes, good catch - I've added conversion to dp